src/scene/Scene.cpp

```diff
@@ -136,7 +147,7 @@ OptixTraversableHandle Scene::buildAS()
 		    .sbtOffset = static_cast<unsigned int>(idx), // NOTE: this assumes a single SBT record per GAS
 		    .visibilityMask = 255,
 		    .flags = OPTIX_INSTANCE_FLAG_DISABLE_ANYHIT,
-		    .traversableHandle = entity->mesh->getGAS(getStream()),
+		    .traversableHandle = gasBuilderForEntities[entity]->getGAS(),
```

Comment:
It works the same as before under the hood. 

Before this PR, meshes that were supposed to be animated could not be shared between entities (because vertices were updated in that mesh object). So it was the Plugin's responsibility to duplicate those meshes.

In this PR, we introduced an Animator that holds animated vertices. It enables mesh sharing for animated entities because the responsibility of duplicating GAS was moved to the RGL. The animator is bound to the Entity so the mapping is created based on it.

The logic of separating GASes for animated entities is well described in [setupGASForEntities()](https://github.com/RobotecAI/RobotecGPULidar/blob/feature/mesh-skinning/src/scene/Scene.cpp#L198) method.

Reply:
Clear!